docs/developer-guide.rst

```diff
@@ -0,0 +1,629 @@
+.. set temporal working directory
+.. jupyter-execute::
+    :hide-code:
+
+    import os
+    import audeer
+
+    _cwd_root = os.getcwd()
+    _tmp_root = audeer.mkdir(os.path.join('docs', 'tmp-developer-guide'))
+    os.chdir(_tmp_root)
+
+
+.. _developer-guide:
+
+Developer guide
+===============
+
+The aim of
+:mod:`audbackend`
+is to provide an
+abstract interface for
+any kind of file storage system.
+Even those,
+that have not been
+invented yet :)
+
+This tutorial is divided
+into three parts.
+Under :ref:`register-a-backend`,
+we show how existing or new backends
+are assigned to names,
+that are used by
+:func:`audbackend.create`
+and
+:func:`audbackend.access`.
+In :ref:`develop-new-interface`,
+we show how to create an interface
+that manages user content.
+Under :ref:`develop-new-backend`,
+we take a deep dive
+and develop a backend
+that stores files into
+a SQLite_ database.
+
+
+.. _register-a-backend:
+
+Register a backend
+------------------
+
+Backends are referred to by names.
+The current active names can be listed by:
```

Comment:
```suggestion
A backend can host
an arbitrary number of repositories.
Repositories created or accessed
in the current session
can be listed with:
```

Reply:
I am not sure if "session" is the best term here...

Reply:
Interesting, this fits into our discussion in https://github.com/audeering/audbackend/pull/176#discussion_r1467692547.

`audbackend.available()` is of cause very blind to already existing repositories. E.g. you can get back an empty list, but you might still get an error when ying to create a repo, but then it turns out the repo exists already.
And on the other hand, a repo becomes part of the list returned by `audbackend.available()` after you have accessed it. But then the information you get from `audbackend.available()` is zero.

But yes, for the current behavior of `audbackend.available()` your wording seems to be fine with me.